src/routes/v2/address.ts

```diff
@@ -141,70 +136,75 @@ async function detailsBulk(
     }
 
     // Enforce no more than 20 addresses.
-    if (addresses.length > 20) {
-      res.json({
+    if (addresses.length > FREEMIUM_INPUT_SIZE) {
+      res.status(400)
+      return res.json({
         error: "Array too large. Max 20 addresses"
       })
     }
 
     logger.debug(`Executing address/details with these addresses: `, addresses)
 
-    // stub response object
-    let returnResponse: IResponse = {
-      status: 100,
-      json: {
-        error: ""
-      }
-    }
+    // Validate each element in the address array.
+    for(let i=0; i < addresses.length; i++) {
+      const thisAddress = addresses[i]
 
-    // Loop through each address and creates an array of insight requests to call in parallel
-    addresses = addresses.map(async (address: any, index: number) => {
       // Ensure the input is a valid BCH address.
       try {
-        BITBOX.Address.toLegacyAddress(address)
```

Comment:
This `map` function was attempting to do data validating *and* return a promise. This is the block that was causing the `fakeurl` `UnhandledPromiseRejectionWarning` errors in the unit tests because there is no error handler for a rejected promise inside the `map` function.

Ultimately, this function wasn't even handling Promises because of the `await` in `detailsFromInsight()`.

To fix this, I moved the validation out and into its own `for` loop. Then handle the processing of the Promises separately. This makes the code easier to read and reason about, and made it actually work as intended.